src/modal_dialog/index.ts

```diff
@@ -0,0 +1,264 @@
+      //@ts-ignore
+      content: isString(cnt) ? createText(cnt) : cnt.get ? cnt.get(0) : cnt,
+      close: () => this.close(),
+    };
+  }
+
+  postRender(view: ModalView) {
+    //@ts-ignore
+    const el = view.model.config.el || view.el;
+    const res = this.render();
+    res && res.appendTo(el);
+  }
+
+  /**
+   * Open the modal window
+   * @param {Object} [opts={}] Options
+   * @param {String|HTMLElement} [opts.title] Title to set for the modal
+   * @param {String|HTMLElement} [opts.content] Content to set for the modal
+   * @param {Object} [opts.attributes] Updates the modal wrapper with custom attributes
+   * @returns {this}
+   * @example
+   * modal.open({
+   *   title: 'My title',
+   *   content: 'My content',
+   *   attributes: { class: 'my-class' },
+   * });
+   */
+  open(opts: any = {}) {
+    const attr = opts.attributes || {};
+    opts.title && this.setTitle(opts.title);
+    opts.content && this.setContent(opts.content);
+    this.model.set('attributes', attr);
+    this.model.open();
+    this.modal && this.modal.updateAttr(attr);
+    return this;
+  }
+
+  /**
+   * Close the modal window
+   * @returns {this}
+   * @example
+   * modal.close();
+   */
+  close() {
+    this.model.close();
+    return this;
+  }
+
+  /**
+   * Execute callback when the modal will be closed.
+   * The callback will be called one only time
+   * @param {Function} clb Callback to call
+   * @returns {this}
+   * @example
+   * modal.onceClose(() => {
+   *  console.log('The modal is closed');
+   * });
+   */
+  onceClose(clb: EventHandler) {
+    this.em.once('modal:close', clb);
+    return this;
+  }
+
+  /**
+   * Execute callback when the modal will be opened.
+   * The callback will be called one only time
+   * @param {Function} clb Callback to call
+   * @returns {this}
+   * @example
+   * modal.onceOpen(() => {
+   *  console.log('The modal is opened');
+   * });
+   */
+  onceOpen(clb: EventHandler) {
+    this.em.once('modal:open', clb);
+    return this;
+  }
+
+  /**
+   * Checks if the modal window is open
+   * @returns {Boolean}
+   * @example
+   * modal.isOpen(); // true | false
+   */
+  isOpen() {
+    return !!this.model.get('open');
+  }
+
+  /**
+   * Set the title to the modal window
+   * @param {string | HTMLElement} title Title
+   * @returns {this}
+   * @example
+   * // pass a string
+   * modal.setTitle('Some title');
+   * // or an HTMLElement
+   * const el = document.createElement('div');
+   * el.innerText =  'New title';
+   * modal.setTitle(el);
+   */
+  setTitle(title: string) {
+    this.model.set('title', title);
+    return this;
+  }
+
+  /**
+   * Returns the title of the modal window
+   * @returns {string | HTMLElement}
+   * @example
+   * modal.getTitle();
+   */
+  getTitle() {
+    return this.model.get('title');
+  }
+
+  /**
+   * Set the content of the modal window
+   * @param {string | HTMLElement} content Content
+   * @returns {this}
+   * @example
+   * // pass a string
```

Comment:
This example is kind of useless after migrating to ts. 

Consider removing it.